### ISSUE TITLE
UnhandledPromiseRejectionWarning: ValidationError

### ISSUE THREAD

**Issue by iraycd** [roboncode/orango#68]
```javascript

// Model.js
  const organizationSchema = new OrganizationSchema(
    {
      email: {
        type: String,
        required: true
      },
      billingEmail: {
        type: String,
        required: true
      },
      name: {
        type: String,
        required: true
      },
      shortName: {
        type: String,
        required: true
      },
      createdBy:  {
        type: String,
        required: true
      }
    },
    {
      strict: true,
      indexes: [
        { type: 'hash', fields: ['active'] },
        { type: 'hash', fields: ['createdBy'] },
        { type: 'hash', fields: ['name'] }
      ]
    }
  )
  const Organization = orango.model('Organization', organizationSchema)

async create(data) {
      return Organization.model.insert(data).return({
        one: true
      })
}
```

```javascript

// Service.js
async create(data) {
  await Model.create({})
}

```

```javascript
 (node:86) UnhandledPromiseRejectionWarning: ValidationError: child "billingEmail" fails because ["billingEmail" is required]. child "name" fails because ["name" is required]. child "shortName" fails because ["shortName" is required]. child "createdBy" fails because ["createdBy" is required]
     at Object.exports.process (/app/node_modules/joi/lib/errors.js:201:19)
     at internals.Object._validateWithOptions (/app/node_modules/joi/lib/types/any/index.js:751:31)
     at internals.Object.validate (/app/node_modules/joi/lib/types/any/index.js:785:21)
     at Schema.validate (/app/node_modules/tangjs/lib/Schema.js:74:16)
     at OrganizationSchema.validate (/app/node_modules/orango/lib/Schema.js:111:28)
     at OrangoModel.validate (/app/node_modules/tangjs/lib/Model.js:92:33)
     at OrangoModel.toObject (/app/node_modules/tangjs/lib/Model.js:115:25)
     at Builder.<anonymous> (/app/node_modules/tangjs/lib/builder.js:18:22)
     at asyncForEach (/app/node_modules/tangjs/lib/builder.js:64:39)
 (node:86) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 6)
```

It occurs both in create as well as update.

**Comment by roboncode** [comment]
This is working as designed. Here is the explanation.

The property `required` accepts three value:
* `true`: required on both create and update
* `"insert"`: required only on insert or save
* `"update"`: required only on update

See snippet https://github.com/roboncode/orango/blob/master/examples/snippets/required.js

If you pass in Model.create({}) with and empty object, it will throw an error(s) when the criteria is not met according to your schema.

**Comment by iraycd** [comment]
I can't catch them as there are `UnhandledPromiseRejectionWarning`.

**Comment by roboncode** [comment]
Use need a try / catch in order to handle the error.

```js
async create(data) {
  try {
     await Model.create({})
  } catch(e) {
     console.log('Error:', e.message)
  } 
}
```

**Comment by iraycd** [comment]
That's not working.

**Comment by roboncode** [comment]
@iraycd I am able to reproduce the issue and have a fix. I am just working on cleanup and then it should be good to go!

**Comment by roboncode** [comment]
Should be fixed in 1.0.0-beta-1.1
```js
async create(data) {
  try {
     await Model.create({})
  } catch(e) {
     console.log('Error:', e.message)
  } 
}
```

**Comment by iraycd** [comment]
Works like a 🍰. Thank you @roboncode
